wallet/src/repository/wallet/mod.rs

```diff
@@ -648,116 +570,96 @@ where
+            let out_ptr: model::OutPtr = input.output_pointer().into();
+
+            if let Some(model::KeyBalance { amount, .. }) = state.utxo_set.get(&out_ptr).cloned() {
+                db_utxo_set.remove(&out_ptr);
+                state.utxo_set.remove(&out_ptr);
+                input_amount = input_amount
+                    .checked_add(amount)
+                    .ok_or_else(|| Error::TransactionBalanceOverflow)?;
+            }
         }
 
-        Ok(())
-    }
-
-    fn _index_transaction_output(
-        &self,
-        state: &mut State,
-        txn_hash: &[u8],
-        out_ptr: model::OutPtr,
-        output: &types::VttOutput,
-        block: &model::BlockInfo,
-    ) -> Result<()> {
-        let pkh = output.pkh;
-        let amount = output.value;
-        let txn_id = state.transaction_next_id;
-        let old_balance = state.balance;
-        let mut batch = self.db.batch();
-
-        if let Some(model::Path { account, .. }) = self.db.get_opt(&keys::pkh(&pkh))? {
-            let new_balance = old_balance
-                .checked_add(amount)
-                .ok_or_else(|| Error::BalanceOverflow)?;
-            let txn_next_id = txn_id
-                .checked_add(1)
-                .ok_or_else(|| Error::TransactionValueOverflow)?;
-            let mut db_utxo_set: model::UtxoSet = self
-                .db
-                .get(&keys::account_utxo_set(account))
-                .unwrap_or_default();
-            let address = pkh.bech32(if self.params.testnet {
-                Environment::Testnet1
-            } else {
-                Environment::Mainnet
-            });
-            let key_balance = model::KeyBalance { pkh, amount };
-
-            match db_utxo_set.insert(out_ptr.clone(), key_balance.clone()) {
-                None => {
-                    log::info!(
-                        "Found transaction to our address {}! Amount: +{} satowits",
-                        address,
-                        amount
-                    );
-                }
-                Some(x) => {
-                    if x != key_balance {
+        let mut output_amount: u64 = 0;
+        for (index, output) in txn.outputs.iter().enumerate() {
+            if let Some(model::Path { .. }) = self.db.get_opt(&keys::pkh(&output.pkh))? {
+                let address = output.pkh.bech32(if self.params.testnet {
+                    Environment::Testnet1
+                } else {
+                    Environment::Mainnet
+                });
+                let key_balance = model::KeyBalance {
+                    pkh: output.pkh,
+                    amount: output.value,
+                };
+
+                let out_ptr = model::OutPtr {
+                    txn_hash: Vec::from(txn_hash),
+                    output_index: index as u32,
+                };
+
+                match db_utxo_set.insert(out_ptr.clone(), key_balance.clone()) {
+                    Some(x) if x == key_balance => {}
```

Comment:
When can this branch happen? When indexing the same transaction twice? Or when some transaction does not change the account balance?